docs/config.rst

```diff
@@ -13,46 +13,89 @@ Defining sessions
 .. autofunction:: nox.session
 
 Nox sessions are configured via standard Python functions that are decorated
-with ``@nox.session`` or start with ``session_``. For example, these are all
-sessions::
+with ``@nox.session``. For example::
 
     import nox
 
-    def session_a(session):
+    @nox.session
+    def b(session):
         pass
 
-    def session_123(session):
+You can also configure sessions to run against multiple Python versions as described in :ref:`virtualenv config` and  parametrize sessions as described in :ref:`parametrized sessions <parametrized>`.
+
+.. _virtualenv config:
+
+Configuring a session's virtualenv
+----------------------------------
+
+By default, Nox will create a new virtualenv for each session using the same interpreter that Nox uses. If you installed Nox using Python 3.6, Nox will use Python 3.6 by default for all of your sessions.
+
+You can tell Nox to use a different Python interpreter/version by specifying the ``python`` argument to ``@nox.session``::
+
+    @nox.session(python='2.7')
+    def tests(session):
         pass
 
-    @nox.session
-    def b(session):
+You can also tell Nox to run your session against multiple Python interpreters. Nox will create a separate virtualenv and run the session for each interpreter you specify. For example, this session will run twice - once for Python 2.7 and once for Python 3.6::
+
+    @nox.session(python=['2.7', '3.6'])
+    def tests(session):
         pass
 
-These are **not**::
+You can use ``nox.python`` to specify even more advanced configuration. For example, if you want to turn off the virtualenv creation::
 
-    def some_session(session):
+    @nox.session(python=nox.python(virtualenv=False))
+    def tests(session):
         pass
 
-    def other_func(session):
+You can also specify that the virtualenv should *always* be reused or change the virtualenv name::
+
+    @nox.session(
+        python=nox.python(
+            python='3.6',
+            virtualenv='custom-name',
+            reuse=True))
```

Comment:
I dislike the stutter that would be involved in this approach. Also, my intuition maintains a cognitive separation between the version of Python being used and the venv configuration as a whole. I would honestly rather see...

```python
@nox.session(python=['2.7', '3.5', '3.6', '3.7'], venv=nox.venv(prefix='...', reuse=True))
```

This has the advantage that you do not have to repeat the virtualenv configuration for every version of Python. It will be a (very) common use case to support 3-4 versions of Python and want virtualenv configuration, and the this involves repeating the `virtualenv=` and `reuse=` arguments over and over for that situation. Separating out the Python versions from the remaining virtualenv configuration is more concise and, I think, more intuitive.

The con to this approach is that you can not have radically distinct virtualenv configurations for different Python versions in the same session; e.g. you can not tell your Python 3.7 virtualenv to be cached and your Python 3.6 virtualenv to be reused. I think that is fine (nobody will want to do that in real life, basically).

If you _really_ think there needs to be different virtualenv configurations for different Python versions, I would rather see stacked `@nox.session` calls, e.g.

```python
@nox.session(python='3.6', virtualenv='tests-3.6')
@nox.session(python='3.7', virtualenv='tests-3.7', reuse=True)
def unit_tests(session):
    pass
```

Reply:
This is sufficient for my goals. :-)

A few questions:

  1. Others had complained about "virtualenv name too long" warnings that nox used to give, especially those who used longer directory structures. Has anything changed, or will that complaint return?

  2. If you _do_ need more granular virtualenv configuration, what is the approach? New keyword arguments here? (That is fine if so.)

Reply:
1. That compliant will return if we do not add the feature in (2).
2. I plan on mirroring py.test's approach for [parametrized ids](https://docs.pytest.org/en/latest/example/parametrize.html#different-options-for-test-ids). Issue filed for it at #96.